src/task.ts

```diff
@@ -12,9 +12,11 @@ export interface Task<TResult, TReason = any> {
   isRunning: boolean;
   isIdle: boolean;
   isCancelled: boolean;
+  isFinished: boolean;
   error?: TReason;
   result?: TResult;
-
+  run(events?: TaskEventNotifications<TResult, TReason>): void;
```

Comment:
You can hide this from the public API by using the `@internal` comment:
https://api-extractor.com/pages/tsdoc/tag_internal/

Reply:
The annotation has to be here, not at the implementation side. The docs are generated based on this.

Needs to be here:
https://github.com/pago/use-concurrent-state/blob/500114f0aa64bd857ff498d6fb6b9b4688865476/src/task.ts#L18

```suggestion
  /**
   * @internal
   */
  run(events?: TaskEventNotifications<TResult, TReason>): void;
```